### ISSUE TITLE
Submit form without any file via jQuery-File-Upload and Asp.net MVC

### ISSUE THREAD

**Issue by PingPongSet** [blueimp/jQuery-File-Upload#3682]
Is it possible to submit a form without any file via jQuery-File-Upload below?

https://github.com/blueimp/jQuery-File-Upload/

I need to make file upload optional, and allow users to submit the form

My current file below:

**Html (ASP.NET View)**

        <form id="fileupload" action="@Url.Action("UploadFiles")" method="POST" enctype="multipart/form-data">
            <div>
                <div class="drop formContentSection" id="dropArea">                    
                    <div class="fileupload-buttonbar">
                        <span class="btn fileinput-button button">
                            <span>Upload FILES</span>
                            <input type="file" name="file" multiple>
                        </span>
                        <span class="fileupload-process"></span>
                    </div>
    
                    
                    <table role="presentation" class="table table-striped">
                        <tbody class="files">
                        </tbody>
                    </table>
                </div>
    
            </div>
    
                <div class="fileupload-buttonbar">
                    <button type="submit" class="start">SUBMIT</button>
                </div>  
        </form>
    	
    	
    	
    <script id="template-upload" type="text/x-tmpl">
        {% for (var i=0, file; file=o.files[i]; i++) { %}
        <tr class="template-upload fade{%=o.options.loadImageFileTypes.test(file.type)?' image':''%}">
            <td>
                <p class="name">{%=file.name%}</p>
                <strong class="error text-danger"></strong>
            </td>
            <td>
                <p class="size">Processing...</p>
                <div class="progress progress-striped active" role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0"><div class="progress-bar progress-bar-success" style="width:0%;"></div></div>
            </td>
            <td>
                {% if (!i && !o.options.autoUpload) { %}
                <button class="btn btn-primary start hidden" disabled>
                    <span>Start</span>
                </button>
                {% } %}
    
                {% if (!i) { %}
                <button class="btn btn-warning cancel btn-sharp">     
                    <span>Cancel</span>
                </button>
                {% } %}
            </td>
        </tr>
        {% } %}
    </script>
    
    
    <script id="template-download" type="text/x-tmpl">
        {% for (var i=0, file; file=o.files[i]; i++) { %}
        <tr class="template-download fade{%=file.thumbnailUrl?' image':''%}">
            <td>
                <p class="name">
                    {% if (file.url) { %}
                    <a href="{%=file.url%}" title="{%=file.name%}" download="{%=file.name%}" {%=file.thumbnailUrl?'data-gallery':''%}>{%=file.name%}</a>
                    {% } else { %}
                    <span>{%=file.name%}</span>
                    {% } %}
                </p>
                {% if (file.error) { %}
                <div><span class="label label-danger">Error</span> {%=file.error%}</div>
                {% } %}
            </td>
            <td>
                <span class="size">{%=o.formatFileSize(file.size)%}</span>
            </td>
            <td>
                {% if (file.deleteUrl) { %}
                <button class="btn btn-danger delete btn-sharp" data-type="{%=file.deleteType%}" data-url="{%=file.deleteUrl%}" {% if (file.deleteWithCredentials) { %} data-xhr-fields='{"withCredentials":true}' {% } %}>
                    <span>Delete</span>
                </button>
                {% } else { %}
                <button class="btn btn-warning cancel">
                    <span>Cancel</span>
                </button>
                {% } %}
            </td>
        </tr>
        {% } %}
    </script>
	
**JavaScript**

    	$(function () {
        'use strict';
    
        $('#fileupload').fileupload(
            {
                dataType: 'json',
                acceptFileTypes: /(\.|\/)(gif|jpe?g|png)$/i,
                singleFileUploads: false,
                autoUpload: false
            });
    
        $('#fileupload').addClass('fileupload-processing');
        $.ajax({
            url: $('#fileupload').fileupload('option', 'url'),
            dataType: 'json',       
            context: $('#fileupload')[0]
        })
            .always(function () {
                $(this).removeClass('fileupload-processing');
            })
            .done(function (result) {
                $(this)
                    .fileupload('option', 'done')
                    .call(this, $.Event('done'), { result: result });
            });
    });

**Server side (Controller)**

      [HttpPost]
            public async Task<JsonFiles> UploadFiles(IFormFile[] file)  //it contains one element. I want all elements in one go
            {
    ...
    }

https://stackoverflow.com/questions/23215608/jquery-file-upload-without-file

**Comment by blueimp** [comment]
Hi @PingPongSet,

my recommendation is to make use of the FormData API to retrieve the form input values and send it via `$.ajax`.

Since you've also posted on StackOverflow, there is an older post with some suggested solutions that might work:
https://stackoverflow.com/questions/21760757/blueimp-jquery-file-upload-how-do-i-submit-form-without-files-attached

**Comment by PingPongSet** [comment]
Hi @blueimp ,

> "my recommendation is to make use of the FormData API to retrieve the form input values and send it via $.ajax."

Is there code example for this?

> Since you've also posted on StackOverflow, there is an older post with some suggested solutions that might work:
> https://stackoverflow.com/questions/21760757/blueimp-jquery-file-upload-how-do-i-submit-form-without-files-attached

There is no accepted answers. I have looked at it, but I am new to this plugin, so am not sure which one will work for me.

**Comment by blueimp** [comment]
Here's a code example for using jQuery's `$.ajax` with `FormData`, again from StackOverflow:
https://stackoverflow.com/a/21045034

Even if there's no accepted answer, I recommend you dig into the answers, try them out, inspect their output and play around with different inputs until you gain an understanding that allows you to adapt it to your own solution.

**Comment by blueimp** [comment]
Closed as support request (see #2190).
